### PR TITLE
Quinilo Notification API

### PR DESCRIPTION
**Pull request by mcalmer**
## What does this PR change?

Added api calls getNotifications, makeNotificationRead, makeAllNotificationsRead, deleteNotification to UserNotificationHandler

Successor of https://github.com/uyuni-project/uyuni/pull/8668

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- API documentation added: please review the Wiki page [Writing Documentation for the API](https://github.com/uyuni-project/uyuni/wiki/Writing-documentation-for-the-API) if you have any changes to API documentation.

- [x] **DONE**

## Test coverage
- Unit tests were added

- [x] **DONE**

## Links

Issue(s): https://github.com/uyuni-project/uyuni/issues/5540

- [x] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

# Before you merge

Check [How to branch and merge properly](https://github.com/uyuni-project/uyuni/wiki/How-to-branch-and-merge-properly)!

### CODE REVIEW

**Comment by mcalmer** [comment]
I have fixed the last minor problems. Can be merged when master branch gets unlocked

File: java/code/src/com/redhat/rhn/frontend/xmlrpc/user/UserNotificationsHandler.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+public class UserNotificationsHandler extends BaseHandler {
+    private static final Logger LOG = LogManager.getLogger(UserNotificationsHandler.class);
+
+    /**
+     * @param user The current user
+     * @param unread Unread notifications
+     * @return Returns a list of notifications
+     * @apidoc.doc Get all notifications from a user.
+     * @apidoc.param #session_key()
+     * @apidoc.param #param_desc("boolean", "unread", "Read notifications.")
```
Comment (cbbayburt) [inline java/code/src/com/redhat/rhn/frontend/xmlrpc/user/UserNotificationsHandler.java:40]:
```suggestion
     * @apidoc.param #param_desc("boolean", "unread", "If true, return only unread notifications.")
```

File: java/code/src/com/redhat/rhn/frontend/xmlrpc/user/UserNotificationsHandler.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        return notifications;
+    }
+
+    /**
+     * @return Returns 1 if successful
+     * @param user The current user
+     * @param notifications Notification list
+     * @apidoc.doc Set notifications of the given user as read
+     * @apidoc.param #session_key()
+     * @apidoc.param #param_desc("Collection", "notifications", "The target notification.")
```
Comment (cbbayburt) [inline java/code/src/com/redhat/rhn/frontend/xmlrpc/user/UserNotificationsHandler.java:65]:
```suggestion
     * @apidoc.param #array_single_desc("int", "notifications", "The list of notification IDs to set as 'read'.")
```

File: java/code/src/com/redhat/rhn/frontend/xmlrpc/user/UserNotificationsHandler.java [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+        return 1;
+    }
+
+    /**
+     * @param notifications The notifications to delete
+     * @param user The current user
+     * @return int number of deleted notifications
+     * @apidoc.doc Deletes multiple notifications
+     * @apidoc.param #session_key()
+     * @apidoc.param #param_desc("collection", "notifications", "List of notification ids.")
```
Comment (cbbayburt) [inline java/code/src/com/redhat/rhn/frontend/xmlrpc/user/UserNotificationsHandler.java:100]:
```suggestion
     * @apidoc.param #array_single_desc("int", "notifications", "The list of notification IDs to delete.")
```

**Review by cbbayburt — CHANGES_REQUESTED** [review]
Overall LGTM.

I commented some quick fixes for the documentation.

**Pull request merged**